WalletWasabi/WabiSabi/Client/ArenaClient.cs

```diff
@@ -21,16 +21,15 @@ public class ArenaClient
 		public static readonly ulong ProtocolMaxWeightPerAlice = 1_000ul;
 
 		public ArenaClient(
+			CredentialPool amountCredentialPool,
+			CredentialPool weightCredentialPool,
 			CredentialIssuerParameters amountCredentialIssuerParameters,
 			CredentialIssuerParameters weightCredentialIssuerParameters,
```

Comment:
I would prefer the issuer parameters first because it is the absolute most required parameter, I mean, we could remove the CredentialPool and thing can still work but there is no way the client can work without the issuer parameters.
  
```suggestion
			CredentialIssuerParameters amountCredentialIssuerParameters,
			CredentialIssuerParameters weightCredentialIssuerParameters,
			CredentialPool amountCredentialPool,
			CredentialPool weightCredentialPool,
```

Reply:
what do you think about defaulting to new pool if not passed by making the constructor arg nullable and optional? it would only really affect the test boilerplate, not sure how I feel about that

Reply:
I changed the order. I think defaulting makes no sense. Production code will always need a separate pool in order to pass it to multiple places. We can have a factory for tests but even there a separate creation is needed I would make any effort.